File: baking/src/main/java/de/pacheco/bakingapp/utils/SimpleIdlingResource.java
package de.pacheco.bakingapp.utils;

import androidx.annotation.Nullable;
import androidx.test.espresso.IdlingResource;

import java.util.concurrent.atomic.AtomicBoolean;

public class SimpleIdlingResource implements IdlingResource {
    AtomicBoolean isIdle = new AtomicBoolean(true);
    @Nullable
    private volatile ResourceCallback callback;

    @Override
    public String getName() {
        return getClass().getName();
    }

    @Override
    public boolean isIdleNow() {
        return isIdle.get();
    }

    @Override
    public void registerIdleTransitionCallback(ResourceCallback callback) {
        this.callback = callback;
    }

    public void setIdleState(boolean isIdleNow) {
        isIdle.set(isIdleNow);
        if (callback != null && isIdleNow) {
            //noinspection ConstantConditions
            callback.onTransitionToIdle();
        }
    }
}

File: baking/src/main/java/de/pacheco/bakingapp/utils/SimpleIdlingResource.kt
package de.pacheco.bakingapp.utils

import androidx.test.espresso.IdlingResource
import java.util.concurrent.atomic.AtomicBoolean

class SimpleIdlingResource : IdlingResource {
    var isIdle = AtomicBoolean(true)

    @Volatile
    private var callback: IdlingResource.ResourceCallback? = null

    override fun getName(): String {
        return javaClass.name
    }

    override fun isIdleNow(): Boolean {
        return isIdle.get()
    }

    override fun registerIdleTransitionCallback(callback: IdlingResource.ResourceCallback) {
        this.callback = callback
    }

    fun setIdleState(isIdleNow: Boolean) {
        isIdle.set(isIdleNow)
        if (callback != null && isIdleNow) {
            callback!!.onTransitionToIdle()
        }
    }
}